libs/agno/agno/models/google/gemini.py

```diff
@@ -1083,3 +1094,613 @@ def _get_metrics(self, response_usage: GenerateContentResponseUsageMetadata) ->
             metrics.provider_metrics = {"traffic_type": response_usage.traffic_type}
 
         return metrics
+
+    def create_file_search_store(self, display_name: Optional[str] = None) -> Any:
+        """
+        Create a new File Search store.
+
+        Args:
+            display_name: Optional display name for the store
+
+        Returns:
+            FileSearchStore: The created File Search store object
+        """
+        config: Dict[str, Any] = {}
+        if display_name:
+            config["display_name"] = display_name
+
+        try:
+            if config:
+                store = self.get_client().file_search_stores.create(config=config)  # type: ignore[arg-type]
+            else:
+                store = self.get_client().file_search_stores.create()
+            log_info(f"Created File Search store: {store.name}")
+            return store
+        except Exception as e:
+            log_error(f"Error creating File Search store: {e}")
+            raise
+
+    async def async_create_file_search_store(self, display_name: Optional[str] = None) -> Any:
+        """
+        Args:
+            display_name: Optional display name for the store
+
+        Returns:
+            FileSearchStore: The created File Search store object
+        """
+        config: Dict[str, Any] = {}
+        if display_name:
+            config["display_name"] = display_name
+
+        try:
+            if config:
+                store = await self.get_client().aio.file_search_stores.create(config=config)  # type: ignore[arg-type]
+            else:
+                store = await self.get_client().aio.file_search_stores.create()
+            log_info(f"Created File Search store: {store.name}")
+            return store
+        except Exception as e:
+            log_error(f"Error creating File Search store: {e}")
+            raise
+
+    def list_file_search_stores(self, page_size: int = 100) -> List[Any]:
```

Comment:
We can merge this with `get_file_search_store`? These are too many methods added to the class